gatsby-node.js

```diff
@@ -1,31 +1,38 @@
-const { createFilePath } = require('gatsby-source-filesystem')
+const {createFilePath} = require('gatsby-source-filesystem')
+const _ = require('lodash')
 
-exports.onCreateNode = ({ node, actions, getNode }) => {
-  const { createNodeField } = actions
+exports.onCreateNode = ({node, actions, getNode}) => {
+  const {createNodeField} = actions
 
   if (node.internal.type === 'Mdx') {
-    const value = createFilePath({ node, getNode })
-
+    const value = createFilePath({node, getNode})
     createNodeField({
       name: 'slug',
       node,
-      value: `${value}`,
+      value: _.get(node, 'internal.frontmatter.slug', value),
     })
   }
 }
 const path = require('path')
 
-exports.createPages = ({ graphql, actions }) => {
-  const { createPage } = actions
+exports.createPages = ({graphql, actions}) => {
+  const {createPage} = actions
   return new Promise((resolve, reject) => {
     resolve(
       graphql(
         `
           {
-            allMdx(sort: { order: ASC, fields: fields___slug }) {
+            allMdx(sort: {order: ASC, fields: fields___slug}) {
               edges {
                 node {
                   id
+                  parent {
+                    ... on File {
+                      name
+                      sourceInstanceName
```

Comment:
this let's us inspect for different types of files based on the name given to the filesystem source.